macros/src/main/scala/pureconfig/Derivation.scala

```diff
@@ -48,17 +48,32 @@ class DerivationMacros(val c: whitebox.Context) extends LazyContextParser with M
     // but we may be unable to parse `Lazy` trees)
     val isHeadImplicitADerivation = c.openImplicits.headOption.exists(_.pre =:= typeOf[Derivation.type])
 
-    // check the `-Xmacro-settings:materialize-derivations` scalac flag
-    if (!isDerivationEnabled || !isHeadImplicitADerivation) {
-      // when not present, simply render the base `Derivation` constructor with `implcitly[A]`. This results in the same
-      // behavior as without `Derivation` (apart from the extra wrapper).
-      q"_root_.pureconfig.Derivation.Successful(_root_.scala.Predef.implicitly[${weakTypeOf[A]}])"
+    // check if the materialization was called explicitly, in which case we want to have the nicer compiler error
+    // messages
+    val isMaterializationExplicitCall = {
+      val firstMacroCall = c.enclosingMacros.reverse.find(c => c.prefix.tree.tpe =:= c.typeOf[Derivation.type])
+      firstMacroCall.fold(false)(c => !c.openImplicits.exists(_.pre =:= c.typeOf[Derivation.type]))
```

Comment:
Can you rename the lambda parameter to something other than `c`, which can be confused with the outer context?

Also, I think it's enough to check here that `c.openImplicits.nonEmpty`, since once we are in _any_ implicit search, we know that methods could not have been called explicitly, right?

Reply:
Yes. Checking that `openImplicits` is empty in the first derivation is sufficient. I've fixed that in dd7c368.